### PR TITLE
Bcrypt: rename `Password#==` to `Password#verify`

### LINKED ISSUE #7753: Crypto::Bcrypt::Password#== is not commutative
As discussed [in the forums](https://forum.crystal-lang.org/t/using-crypto-password-for-authentication/743?u=asterite), it's a bit unintuitive that given `password : Crypto::Bcrypt::Password` and `string : String`, `password == string` will work but `string == password` won't.

Two solutions:
1. Add an overload `String==#(other : Crypto::Bcrypt::Password)` so it's commutative
2. Use a different method name. Ruby's BCrypt uses `is_password?` as an alias, but `verify`, `check` or `matches?` might be equally good.

I'm not sure which approach is better. I think the second one is better because it's also more type-safe. You can do `password == 1.23` and it will still compile, but it's obviously incorrect. You can even do `password == password` and it will return true, but it makes little sense.

Thoughts?

### PR DESCRIPTION
**Pull request by asterite**
Fixes #7753

### CODE REVIEW

File: src/crypto/bcrypt/password.cr [hunk 1]
```diff
@@ -61,14 +61,19 @@ class Crypto::Bcrypt::Password
...
+  # password.verify("wrong secret") # => false
+  # password.verify("super secret") # => true
   # ```
-  def ==(password : String) : Bool
+  def verify(password : String) : Bool
     hashed_password = Bcrypt.new(password, salt, cost)
     Crypto::Subtle.constant_time_compare(@raw_hash, hashed_password)
   end
 
+  @[Deprecated("Use Crypto::Bcrypt::Password#verify")]
```
Comment (bcardiff) [inline src/crypto/bcrypt/password.cr]:
```suggestion
  @[Deprecated("Use `Crypto::Bcrypt::Password#verify`")]
```

**Comment by oprypin** [comment]
Maybe not breaking-change if it's just deprecated

**Comment by bcardiff** [comment]
@oprypin I know, but we agree that at least for the time being we are tagged them as breaking for better visibility in the release notes.

**Comment by Sija** [comment]
I'd still suggest to rename the method to `verify?` to make it obvious it's returning a `Bool`.

**Comment by asterite** [comment]
@Sija But `verify?` isn't proper English, same reason we have `includes?` and not `include?`. I think `verify` is fine. I would have liked `matches?` but others agreed with using `verify` or `verify?` (but `verify?` is inconsistent to me).

**Pull request merged**